### ISSUE TITLE
ENet training on Cityscapes

### ISSUE THREAD

**Issue by akshaykulkarni07** [akshaykvnit/autonomous-delivery-robot#11]
ENet has been implemented in PyTorch (Python2) and tested successfully on Jetson TX1 (#1). Further tasks: 
- [x] Create a small subset of Cityscapes for testing purposes.
- [x] Implement metrics for monitoring. Using IoU (intersection over union).
- [x] Implement save best feature during training based on metric.
- [x] Train model.

**Comment by akshaykulkarni07** [comment]
At first, training around 50 epochs on Cityscapes was not giving good results on the local images. Having tried many different hyperparameters and also some learning rate schedulers, finally started checking the outputs at early stages and found that early outputs were promising. This indicates that training too much on Cityscapes does not help the network to generalize for our local usecase.

Further, while testing we get as output the probabilities that each pixel represents a road. So, we need to set a threshold between 0 and 1 above which pixel is labeled road. Currently, setting it to 0.3 gives decent results (Note: too high threshold will cause misclassification i.e. obstacles being labeled road, while too low threshold will not give the complete road). We can play around with this parameter as soon as we get more video samples.

The best weights file and sample results are [saved in Google Drive](https://drive.google.com/open?id=1iAGBhPr8EuetmvGqnZJG0ZdWnMSxFpnM).
